va/caa.go

```diff
@@ -160,7 +148,29 @@ func (va *ValidationAuthorityImpl) parallelCAALookup(ctx context.Context, name s
 	return results
 }
 
-func (va *ValidationAuthorityImpl) getCAASet(ctx context.Context, hostname string) (*CAASet, string, error) {
+// selectCAA picks the relevant CAA resource record set to be used, i.e. the
+// set for the "closest parent" of the FQDN in question, including the domain
+// itself. If the first non-empty RR set we find encountered an error during
+// lookup, we must assume that there could have been real records hidden by the
+// error, and propagate that error upwards.
```

Comment:
If we encountered an error, it's not a non-empty RRSet, right?

```suggestion
// itself. If we encountered an error for a lookup before we found a successful, non-empty response,
// assume there could have been real records hidden by it, and return that error.
```